src/realm/alloc_slab.cpp

```diff
@@ -388,11 +387,6 @@ SlabAlloc::FreeBlock* SlabAlloc::allocate_block(int size)
     if (remaining)
         push_freelist_entry(remaining);
     REALM_ASSERT_EX(size_from_block(block) >= size, size_from_block(block), size, get_file_path_for_assertions());
-    auto block_before = bb_before(block);
```

Comment:
I don't thin we should delete this. It is good for catching memory corruption.

Reply:
Ok let's keep them in debug mode, throwing an exception seems  too much.